### PR TITLE
fix(dynamic_scale_down_mgr): do not attempt to scale down enterprise clusters

### PR DESCRIPTION
**Pull request by machi1990**
## Description

While trying out the fleet manager locally, I observed that KFM attemps to scale down the enterprise cluster as can be observed from the logs.
```
I0322 16:21:16.036258 1 dynamic_scale_down_mgr.go:114] evaluating dynamic scale down for cluster "23ruvhn0o1sqspso2sce69g5ghbamjoo"
I0322 16:21:16.036275 1 dynamic_scale_down_mgr.go:315] skipping candidate cluster with cluster id "23ruvhn0o1sqspso2sce69g5ghbamjoo"
I0322 16:21:16.036284 1 dynamic_scale_down_mgr.go:121] data plane scale down need detected for cluster "23ruvhn0o1sqspso2sce69g5ghbamjoo"
I0322 16:21:16.036289 1 dynamic_scale_down_mgr.go:370] scale down running in dryRun mode. No action is taken for cluster with cluster id "23ruvhn0o1sqspso2sce69g5ghbamjoo".

```

`23ruvhn0o1sqspso2sce69g5ghbamjoo` is an enterprise cluster that I've in the DB.

This should never happen and we should always skip enterprise cluster from scaling down.
 
## Verification Steps

The added unit tests should pass

## Checklist (Definition of Done)

- [ ] All acceptance criteria specified in JIRA have been completed
- [ ] Unit and integration tests added that prove the fix is effective or the feature works (tested against emulated and non-emulated OCM environment)
- [ ] Documentation added for the feature
- [ ] CI and all relevant tests are passing
- [ ] Code Review completed
- [ ] Verified independently by reviewer
- [ ] All PR comments are resolved either by addressing them or creating follow up tasks
- [ ] Required metrics/dashboards/alerts have been added (or PR created).
- [ ] Required Standard Operating Procedure (SOP) is added.
- [ ] JIRA has been created for changes required on the client side

### CODE REVIEW

**Review by pb82 — APPROVED** [review]
LGTM, but best if someone more involved in the dynamic scaling work can also review.

File: internal/kafka/internal/workers/cluster_mgrs/dynamic_scale_down_mgr.go [hunk 1]
```diff
@@ -254,6 +252,12 @@ var _ dynamicScaleDownProcessor = &standardDynamicScaleDownProcessor{}
 // 2. Clusters in deprovisioning and cleanup state are excluded, as clusters into those states don't accept kafka instances anymore.
 // 3. Clusters that are still not ready to accept kafka instance are also excluded from the capacity calculation
 func (p *standardDynamicScaleDownProcessor) ShouldScaleDown() (bool, error) {
+	// check if the cluster is an enterprise cluster
+	if p.isEnterpriseCluster() {
```
Comment (miguelsorianod) [inline internal/kafka/internal/workers/cluster_mgrs/dynamic_scale_down_mgr.go:256]: I am wondering if this is the place where we should be concerned about it.
What about directly not returning it from the calculated streaming units? or filtering through the locator appropriately?

File: internal/kafka/internal/workers/cluster_mgrs/dynamic_scale_down_mgr.go [hunk 2]
```diff
@@ -163,16 +163,14 @@ func (m *DynamicScaleDownManager) createAMapOfProcessedClusters(kafkaStreamingUn
 	var processedClusters map[string]processed = map[string]processed{}
 
 	for i, kafkaStreamingUnitCountPerCluster := range kafkaStreamingUnitCountPerClusterList {
-		if kafkaStreamingUnitCountPerCluster.ClusterType != api.ManagedDataPlaneClusterType.String() {
```
Comment (miguelsorianod) [inline internal/kafka/internal/workers/cluster_mgrs/dynamic_scale_down_mgr.go:166]: Why is this removed?

File: internal/kafka/internal/workers/cluster_mgrs/dynamic_scale_down_mgr.go
[the remark this answers is not in the document]
Comment (machi1990) [reply internal/kafka/internal/workers/cluster_mgrs/dynamic_scale_down_mgr.go:1]: scale up is handled already using the locator

File: internal/kafka/internal/workers/cluster_mgrs/dynamic_scale_down_mgr.go (hunk 1 above)
Comment (machi1990) [reply internal/kafka/internal/workers/cluster_mgrs/dynamic_scale_down_mgr.go:256]: all clusters are returned in the streaming unit calculations: enterprise and managed ones.
There is no notion of locator we could ultimately move the logic here https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/aedf21a02b82289bff00df4cd337fd83ec6d2d26/internal/kafka/internal/workers/cluster_mgrs/dynamic_scale_down_mgr.go#L89 if people feel like that's the best place. I do not mind either way

File: internal/kafka/internal/workers/cluster_mgrs/dynamic_scale_down_mgr.go (hunk 2 above)
Comment (machi1990) [reply internal/kafka/internal/workers/cluster_mgrs/dynamic_scale_down_mgr.go:166]: We needed to construct the struct needed for `ShouldScaleDown` evaluation for enterprise clusters. 
With this check, all enterprise clusters were skipped from that struct - which had another ripple effect of not considering enterprise clusters as being processed in https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/aedf21a02b82289bff00df4cd337fd83ec6d2d26/internal/kafka/internal/workers/cluster_mgrs/dynamic_scale_down_mgr.go#L92

File: internal/kafka/internal/workers/cluster_mgrs/dynamic_scale_down_mgr.go (hunk 1 above)
Comment (miguelsorianod) [reply internal/kafka/internal/workers/cluster_mgrs/dynamic_scale_down_mgr.go:256]: I think having some concept of locator as in scale up might be nice. I haven't looked in detail how good it would fit in the current scale down structure though. Could you review if it makes sense similarly to scale up?

Comment (machi1990) [reply internal/kafka/internal/workers/cluster_mgrs/dynamic_scale_down_mgr.go:256]: It could probably make sense and the existing logic in  https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/aedf21a02b82289bff00df4cd337fd83ec6d2d26/internal/kafka/internal/workers/cluster_mgrs/dynamic_scale_down_mgr.go#L87 which groups similar streaming unit counts of a cluster (a cluster given by cluster id could appear more than once if it support several kafka instance type) and then used here https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/aedf21a02b82289bff00df4cd337fd83ec6d2d26/internal/kafka/internal/workers/cluster_mgrs/dynamic_scale_down_mgr.go#L91-L112 could be refactored onto a locator but I prefer to defer that extra refactoring work from this PR.

Comment (miguelsorianod) [reply internal/kafka/internal/workers/cluster_mgrs/dynamic_scale_down_mgr.go:256]: :+1:

**Review by miguelsorianod — APPROVED** [review]

**Comment by machi1990** [comment]
Thanks for the review @miguelsorianod @pb82
